modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/PrecisionTests.java

```diff
@@ -186,6 +197,49 @@ public void testXContentRoundtrip() throws IOException {
         assertEquals(testItem, parsedItem);
         assertEquals(testItem.hashCode(), parsedItem.hashCode());
     }
+    
+    public void testSerialization() throws IOException {
+        Precision original = createTestItem();
+        Precision deserialized = RankEvalTestHelper.copy(original, Precision::new);
+        assertEquals(deserialized, original);
+        assertEquals(deserialized.hashCode(), original.hashCode());
+        assertNotSame(deserialized, original);
+    }
+
+    public void testEqualsAndHash() throws IOException {
+        Precision testItem = createTestItem();
+        RankEvalTestHelper.testHashCodeAndEquals(testItem, mutateTestItem(testItem),
+                RankEvalTestHelper.copy(testItem, Precision::new));
+    }
+
+    private Precision mutateTestItem(Precision testItem) {
+        boolean ignoreUnlabeled = testItem.getIgnoreUnlabeled();
+        int relevantThreshold = testItem.getRelevantRatingThreshold();
+        
+        int mutate = randomIntBetween(0, 1);
+        switch (mutate) {
+            case 0:
+            {
+                ignoreUnlabeled = ! ignoreUnlabeled;
+                break;
+            }
+            case 1:
+            {
+                int mutation = randomIntBetween(0, 10);
+                while (mutation == relevantThreshold) {
```

Comment:
There's a nice helper for this kind of random value creation in `ESTestCase#randomValueOtherThan(T input, Supplier<T> randomSupplier`). The second argument is a function that generates a random value, the first argument is a value that is supposed to be excluded from the range of possible random values. This might simplify this mutate code (and others in this PR) quiet a bit.